src/emu/debug/debughlp.cpp

```diff
@@ -1586,3 +1602,91 @@ const char *debug_get_help(const char *tag)
 			msglen += sprintf(&ambig_message[msglen], "  help %s?\n", static_help_list[i].tag);
 	return ambig_message;
 }
+
+void debug_format_chapter_html(int index)
+{
+	/* fetch the string */
+	debug_buffer = static_help_list[index].help;
+
+	/* substitute what html can't ignore */
+	strreplace(debug_buffer, "<", "&lt;");
+
+	/* iterate through all lines */
+	for (size_t linebreak = 0; ; )
+	{
+		/* find line break */
+		linebreak = debug_buffer.find_first_of('\n', linebreak);
+
+		/* check validity and advance */
+		if (linebreak >= debug_buffer.length() || linebreak++ == std::string::npos)
+			break;
+
+		/* check if the new line starts with a space */
+		if (debug_buffer[linebreak] != ' ')
+		{
+			/* close the tag and advance */
+			debug_buffer.insert(linebreak, html_preclose);
+			linebreak += html_preclose.length();
+
+			/* find the end of the line */
+			linebreak = debug_buffer.find_first_of('\n', linebreak);
+
+			/* check validity again */
+			if (linebreak >= debug_buffer.length() || linebreak == std::string::npos)
+				break;
+
+			/* open a new tag at the line end and advance */
+			debug_buffer.insert(linebreak, html_preopen);
+			linebreak += html_preopen.length();
+		}
+	}
+}
+
+bool debug_generate_html()
```

Comment:
This function leaks the open file if any of the `fprintf` calls fails - that's not acceptable error handling.